admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,319 @@
+| None
+| Monitor ABRT log messages and report them further. ABRT (Automatic Bug Report Tool) is health monitoring daemon able 
+to catch kernel problems as well as application crashes of various kinds occurred on the host.
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor for node-problem-detector to invoke and check various node problems with user defined check scripts.
+
+|===
+
+//https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#node-problem-detector
+The node problem detector supports file-based kernel logging. However, it is easy to extend it to xref:admin-guide-node-problem-detector-log[support other log format].
+
+[NOTE]
+====
+The node problem detector does not detect problems reported by log tools, such as journald or rsyslogd.
+====
+
+[[admin-guide-node-problem-detector-run]]
+== Running the Node Problem Detector
+
+To run the Node Problem Detector first create a xref:../dev_guide/daemonsets#dev-guide-daemonsets[daemonset] object. For {product-title} 
+Google Compute Engine (GCE) infrastructures, see xref:[Running the Node Problem Detector in a GCE Cluster].
+
+When xref:../dev_guide/daemonsets.adoc#dev-guide-daemonsets[creating daemonsets], the `nodeSelector` field indicates the nodes on which the daemonset should deploy.
+
+. Copy the following code to a YAML file on your master:
++
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: node-problem-detector-v0.1
+  namespace: kube-system
+  labels:
+    k8s-app: node-problem-detector
+    version: v0.1
+    kubernetes.io/cluster-service: "true"
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: node-problem-detector
+        version: v0.1
+        kubernetes.io/cluster-service: "true"
+    spec:
+      nodeSelector: <1>
+        type: infra
+      hostNetwork: true
+      containers:
+      - name: node-problem-detector
+        image: k8s.gcr.io/node-problem-detector:v0.1
+        securityContext:
+          privileged: true
+        resources:
+          limits: 
+            cpu: "200m"
+            memory: "100Mi"
+          requests:
+            cpu: "20m"
+            memory: "20Mi"
+        volumeMounts:
+        - name: log 
+          mountPath: /log 
+          readOnly: true
+      volumes:
+      - name: log
+        hostPath: <2>
+          path: /var/log/
+----
++
+<1> The node selector that determines on which nodes pod replicas should be deployed
+<2> Specify the location of your cluster log directory.
+
+. Edit the file for your environment. 
+
+. Save the file as *_node-problem-detector.yaml_*.
+
+. Run the following command to start node problem detector: 
++
+----
+oc create -f <your_yaml_file>.yaml
+----
+
+[[admin-guide-node-problem-detector-addon]]
+== Running the Node Problem Detector as an Add-on Pod
+
```

Comment:
Can customers run NPD as an add-on? I don't see references in our docs on Kubernetes Add-ons, Add-on Manger. Maybe we use different wording. I can find out if we support this.